subsys/net/ip/net_shell.c

```diff
@@ -3535,7 +3535,347 @@ static int cmd_net_ipv6(const struct shell *sh, size_t argc, char *argv[])
+	PR("============================================%s\n", extra);
+
+	if (!ipv4) {
+		PR("No IPv4 config found for this interface.\n");
+		return;
+	}
+
+	PR("Type      \tState    \tLifetime (sec)\tAddress\n");
+
+	for (i = 0; i < NET_IF_MAX_IPV4_ADDR; i++) {
+		char remaining_str[sizeof("01234567890")];
+		uint64_t remaining;
+
+		if (!ipv4->unicast[i].is_used ||
+		    ipv4->unicast[i].address.family != AF_INET) {
+			continue;
+		}
+
+		remaining = net_timeout_remaining(&ipv4->unicast[i].lifetime,
+						  k_uptime_get_32());
+
+		if (ipv4->unicast[i].is_infinite) {
+			snprintk(remaining_str, sizeof(remaining_str) - 1,
+				 "infinite");
+		} else {
+			snprintk(remaining_str, sizeof(remaining_str) - 1,
+				 "%u", (uint32_t)(remaining / 1000U));
+		}
+
+		PR("%s  \t%s\t%s    \t%12s/%12s\n",
+		       addrtype2str(ipv4->unicast[i].addr_type),
+		       addrstate2str(ipv4->unicast[i].addr_state),
+		       remaining_str,
+		       net_sprint_ipv4_addr(
+			       &ipv4->unicast[i].address.in_addr),
+		       net_sprint_ipv4_addr(
+			       &ipv4->netmask));
+	}
+}
+#endif /* CONFIG_NET_NATIVE_IPV4 */
+
+static int cmd_net_ipv4(const struct shell *shell, size_t argc, char *argv[])
+{
+	PR("IPv4 support                              : %s\n",
+	   IS_ENABLED(CONFIG_NET_IPV4) ?
+	   "enabled" : "disabled");
+	if (!IS_ENABLED(CONFIG_NET_IPV4)) {
+		return -ENOEXEC;
+	}
+
+#if defined(CONFIG_NET_NATIVE_IPV4)
+	struct net_shell_user_data user_data;
+
+	PR("IPv4 fragmentation support                : %s\n",
+	   IS_ENABLED(CONFIG_NET_IPV4_FRAGMENT) ? "enabled" :
+	   "disabled");
+	PR("Max number of IPv4 network interfaces "
+	   "in the system          : %d\n",
+	   CONFIG_NET_IF_MAX_IPV4_COUNT);
+	PR("Max number of unicast IPv4 addresses "
+	   "per network interface   : %d\n",
+	   CONFIG_NET_IF_UNICAST_IPV4_ADDR_COUNT);
+	PR("Max number of multicast IPv4 addresses "
+	   "per network interface : %d\n",
+	   CONFIG_NET_IF_MCAST_IPV4_ADDR_COUNT);
+
+	user_data.shell = shell;
+	user_data.user_data = NULL;
+
+	/* Print information about address lifetime */
+	net_if_foreach(ip_address_lifetime_cb, &user_data);
+#endif
+
+	return 0;
+}
+
+static int cmd_net_ip_add(const struct shell *shell, size_t argc, char *argv[])
+{
+#if defined(CONFIG_NET_NATIVE_IPV4)
+	struct net_if *iface = NULL;
+	int idx;
+	struct in_addr addr;
+
+	if (argc != 4) {
+		PR_ERROR("Correct usage: net ipv4 add <index> <address> <netmask>\n");
+		return -EINVAL;
+	}
+
+	idx = get_iface_idx(shell, argv[1]);
+	if (idx < 0) {
+		return -ENOEXEC;
+	}
+
+	iface = net_if_get_by_index(idx);
+	if (!iface) {
+		PR_WARNING("No such interface in index %d\n", idx);
+		return -ENOEXEC;
+	}
+
+	if (net_addr_pton(AF_INET, argv[2], &addr)) {
+		PR_ERROR("Invalid address: %s\n", argv[2]);
+		return -EINVAL;
+	}
+
+	net_if_ipv4_addr_add(iface, &addr, NET_ADDR_MANUAL, 0);
+
+	if (net_addr_pton(AF_INET, argv[3], &addr)) {
+		PR_ERROR("Invalid netmask: %s", argv[3]);
+		return -EINVAL;
+	} else {
```

Comment:
Just move `net_if_ipv4_set_netmask(iface, &addr);` out of `else` to make CI happy.

Reply:
Done